lib/src/network/mod.rs

```diff
@@ -204,14 +204,14 @@ pub enum SocketType {
   FrontClient
 }
 
-#[derive(Debug,PartialEq,Eq)]
+#[derive(Debug,PartialEq,Eq,Clone)]
```

Comment:
why do you need to implement `Clone` on all those structs now? The `Backend` is already inside a `Rc<RefCell<..>>`

Reply:
Not a problem for `when there is no`. That was for the unit test "it_should_find_the_backend_with_least_connections" where I clone the backend `Rc::new(RefCell::new(backend_with_least_connection.clone())),`. Do you mind if I do this?

Reply:
I'd prefer you wrap it in a Rc in the test too, then clone that instead of adding those traits just for the test

Reply:
Ok I'll try